_posts/RhymesWithMecca/2016-02-03-BeccasChapter5.md

```diff
@@ -0,0 +1,23 @@
+---
+layout: post
+author: RhymesWithMecca
+title: "Becca's Chapter 5 Exercises"
+---
+
+Here is Exercise 1:
+<iframe src="https://trinket.io/embed/python/79b285a420" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Here is Exercise 2:
+<iframe src="https://trinket.io/embed/python/22b00b018c" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Exercise 1 Thoughts:
+This one took me a while.  First I had to figure out the proper order of the ifs, try/except, and while loop.  I needed to put the if and break statement before turning the input into a float.  I screwed up my indentation in the try conditional.  At one point, it took inputs but always printed the error message in the except loop.
+
+Then I got stuck and went to sleep.  The next day, I tried it again with functions, but it was printing something like <function sum> and I couldn't figure out why.  The weird input messages were still occurring.  I attempted to use global x for the first time, but no dice.
+
+I then looked ahead on the schedule and saw that Chapter 8 is about lists.  This looked promising.  Midway through the chapter, I saw a part about remaking the original summing program and thought that might be useful.  I used that to help me write my program.
+
+The one thing I am still confused about is why you need to make a new name for the input for the user.  I called it x and then had to give it a new name (value).  When I didn't do this, I had the error messages.  I don't remember doing this in the past.
```

Comment:
This worked for me: https://trinket.io/python/04b2b66a4d

Perhaps it was something else?  The thing with a try/except loop is that if you have an unrelated error (like a typo) it will catch it.